dashboard/src/utils/constructDate.js

```diff
@@ -0,0 +1,3 @@
+export const constructNewDate = (date) => {
+  return new Date(`${date.split(":")[0]}T00:00:00`);
```

Comment:
I don't think that this is going to do what you want it to, @shubham-html-css-js:  I believe that JS is going to take that argument as _local time_, and it will apply an offset to it to make it into UTC.  While I can see how you would make that work for the cases under consideration here, I don't think the result will be generalizable.  (That is, this is going to get you into trouble later.)

You need to do something to tell JS that the argument is in UTC and not local time.  One way is to include the timezone indicator in the string, e.g.:
```
  return new Date(`${date.split(":")[0]}T00:00:00Z`);
```
(note the trailing `Z`).

Another way is to adjust the timezone offset explicitly:
```
  const date_UTC = new Date(date)
  date_UTC.setMinutes(-date.getTimezoneOffset())
  return date_UTC
```